modules/core/src/main/scala/doobie/util/fragments.scala

```diff
@@ -45,6 +47,45 @@ object fragments {
     NonEmptyList.fromFoldable(fs).map(nel => notIn(f, nel))
   }
 
+  @inline
+  private def inNotInValues[F[_]: Reducible: Functor, A](
+      fwop: Fragment,
+      fs: F[A],
+      fallback: Fragment
+  )(implicit A: util.Write[A]): Fragment = {
+    if (A.length == 0) fallback
```

Comment:
Thanks this looks good! Small suggestion - I don't think we need/should handle 0-arity products.

Reason:
- We no longer derive instances for case objects so it's really difficult to accidentally pass in a 0-arity product 
- Which means the only other possible case is if user is passing `List[Unit]`, which never makes sense and it's better to generate a bad SQL that fails than silently fallback to TRUE/FALSE.

WDYT?

Reply:
Agree, I don't like it either. I'll update the PR, thx!